deepspeed/pt/deepspeed_cuda.py

```diff
@@ -95,7 +98,8 @@ def __init__(self,
                  gelu_checkpoint=False,
                  adjust_init_range=True,
                  attn_dropout_checkpoint=False,
-                 stochastic_mode=False):
+                 stochastic_mode=False,
+                 huggingface=False):
```

Comment:
I am a little nervous using `huggingface` as a boolean flag because the API gets worse after we support another target replacement.

Since this flag only changes `forward()`'s return type -- `(output, )` instead of `output` -- we could name this flag more specifically like `return_tuple` ? Then in the docstring we state it's for HF compatibility.

Alternatively, we should plan for variations in each transformer implementation and treat this as more of a base class that specific implementations specialize around. From above, the HF specialization could be something like this:

```python
class HFTransformerLayer(TransformerLayer):
    def forward(...):
        output = super().forward(..) # this applies our transformer function
        return (output, ) # wrap in a tuple
```

@tjruwase has some great ideas on formalizing these replacement interfaces in order to simplify adding new types.